crates/sui-storage/src/lib.rs

```diff
@@ -76,74 +81,61 @@ impl FileCompression {
     }
 }
 
-#[derive(Copy, Clone, Debug, Eq, PartialEq, TryFromPrimitive, IntoPrimitive)]
-#[repr(u8)]
-pub enum Encoding {
-    Bcs = 1,
+pub fn compute_sha3_checksum_for_file(file: &mut File) -> Result<[u8; 32]> {
+    let mut hasher = Sha3_256::default();
+    io::copy(file, &mut hasher)?;
+    Ok(hasher.finalize().digest)
 }
 
-pub struct Blob {
-    pub data: Vec<u8>,
-    pub encoding: Encoding,
+pub fn compute_sha3_checksum(source: &std::path::Path) -> Result<[u8; 32]> {
+    let mut file = fs::File::open(source)?;
+    compute_sha3_checksum_for_file(&mut file)
 }
 
-impl Blob {
-    pub fn encode<T: Serialize>(value: &T, encoding: Encoding) -> Result<Self> {
-        let value_buf = bcs::to_bytes(value)?;
-        let (data, encoding) = match encoding {
-            Encoding::Bcs => (value_buf, encoding),
-        };
-        Ok(Blob { data, encoding })
-    }
-    pub fn decode<T: DeserializeOwned>(self) -> Result<T> {
-        let data = match &self.encoding {
-            Encoding::Bcs => self.data,
-        };
-        let res = bcs::from_bytes(&data)?;
-        Ok(res)
-    }
-    pub fn read<R: Read>(rbuf: &mut R) -> Result<Blob> {
-        let len = rbuf.read_varint::<u64>()? as usize;
-        if len == 0 {
-            return Err(anyhow!("Invalid object length of 0 in file"));
+pub fn write<R: Read, W: Write>(reader: &mut R, writer: &mut W) -> Result<()> {
+    let magic = reader.read_u32::<BigEndian>()?;
```

Comment:
@mystenmark - this is the `write` function referenced in the previous comment where compression is taking place

Reply:
renamed to `compress`